sklearn/preprocessing/_data.py

```diff
@@ -944,16 +947,16 @@ def partial_fit(self, X, y=None, sample_weight=None):
         else:
             # First pass
             if not hasattr(self, "scale_"):
-                self.mean_ = 0.0
+                self.mean_ = xp.zeros(1)[0]  # TODO is there a better way?
```

Comment:
How can I create a scalar value of a certain dtype? (or: Does it actually have to be a scalar value?)

Reply:
Why not keep a Python scalar here? Broadcasting Python scalars should work for all Array API implementations, no?

Reply:
If not, we would probably also need to set the `dtype` and the `device` explicitly.

Reply:
this should be solved :)